diff --git a/src/Entity/WorkflowStorage.php b/src/Entity/WorkflowStorage.php
index 8c28242..1666690 100644
--- a/src/Entity/WorkflowStorage.php
+++ b/src/Entity/WorkflowStorage.php
@@ -2,6 +2,7 @@

 namespace Drupal\workflow\Entity;

+use Drupal\Core\Cache\MemoryCache\MemoryCacheInterface;
 use Drupal\Core\Config\Entity\ConfigEntityStorage;
 use Drupal\Core\State\StateInterface;
 use Drupal\Core\Entity\EntityInterface;
@@ -34,11 +35,13 @@ class WorkflowStorage extends ConfigEntityStorage {
    *   The UUID service.
    * @param \Drupal\Core\Language\LanguageManagerInterface $language_manager
    *   The language manager.
+   * @param \Drupal\Core\Cache\MemoryCache\MemoryCacheInterface $memory_cache
+   *   The memory cache backend.
    * @param \Drupal\Core\State\StateInterface $state_service
    *   The state service.
    */
-  public function __construct(EntityTypeInterface $entity_type, ConfigFactoryInterface $config_factory, UuidInterface $uuid_service, LanguageManagerInterface $language_manager, StateInterface $state_service) {
-    parent::__construct($entity_type, $config_factory, $uuid_service, $language_manager);
+  public function __construct(EntityTypeInterface $entity_type, ConfigFactoryInterface $config_factory, UuidInterface $uuid_service, LanguageManagerInterface $language_manager, MemoryCacheInterface $memory_cache, StateInterface $state_service) {
+    parent::__construct($entity_type, $config_factory, $uuid_service, $language_manager, $memory_cache);

     $this->stateService = $state_service;
   }
@@ -52,6 +55,7 @@ class WorkflowStorage extends ConfigEntityStorage {
       $container->get('config.factory'),
       $container->get('uuid'),
       $container->get('language_manager'),
+      $container->get('entity.memory_cache'),
       $container->get('state')
     );
   }
diff --git a/workflow.services.yml b/workflow.services.yml
index fd75fc2..f6fb214 100644
--- a/workflow.services.yml
+++ b/workflow.services.yml
@@ -5,6 +5,6 @@ services:

   workflow.route_subscriber:
     class: Drupal\workflow\Routing\RouteSubscriber
-    arguments: ['@entity.manager']
+    arguments: ['@entity_type.manager']
     tags:
       - { name: event_subscriber }
